### DISCUSSION TITLE
Dynamic Fulfillment

### DISCUSSION THREAD

**Discussion by haveamission** [medusajs/medusa#4355] [Feature Requests]
I am trying to create a custom fulfillment provider for a distributor, and their fulfillment process is dynamic - they generate a quote for shipping from various providers from a given cart.

This doesn't seem to mesh well with Medusa's need for static shipping methods with a fulfillment provider.

I would like the ability to have dynamic shipping options for a fulfillment provider

**Comment by petrpatek** [comment]
I am having the same challenge. @haveamission have you managed to come up with something for a current Medusa version by any chance? I am now sticking only to those that are available region-wide. One more idea I  got is to use the RegionService inside the `getFulfillmentOptions` and fetch the providers from the 3rd party vendor here.

**Comment by haveamission** [comment]
> have you managed to come up with something for a current Medusa version by any chance?

So I actually managed to talk with the Medusa team on this, there's no baseline support for this. That being said, they did point me to the Shippo Medusa integration and they did a pretty similar implementation to what you are describing above.

I have a slightly different implementation (because users are only supposed to be able to select a smaller subset of available options for my client's specific use-case), but I am considering switching to an implementation more similar to Shippo's (and the one you're suggesting).

**Comment by JurgenSchadtler** [comment]
@haveamission did you use the shippo plugin? I saw that the plugin last update was 2 years ago and was trying to make my own implementation but failed doing so.
